src/main/java/io/github/opencubicchunks/cubicchunks/core/server/chunkio/region/ShadowPagingRegion.java

```diff
@@ -74,25 +81,79 @@ private ShadowPagingRegion(FileChannel file, SectorTracker<K> sectorTracker, IHe
 		this.sectorTracker = sectorTracker;
 	}
 
-	@Override public synchronized void writeValue(K key, ByteBuffer value) throws IOException {
+	/**
+	 * Writes an entry without updating the region headers on disk.
+	 * @return whether or not {@link #file} needs to be flushed before the headers are be updated
+	 */
+	protected boolean writeValueFirst(K key, ByteBuffer value) throws IOException {
 		if (value == null) {
 			this.sectorTracker.removeKey(key);
-			updateHeaders(key);
+			return false; //the data isn't actually changed on-disk, so we don't need to flush the channel before updating the headers
```

Comment:
removing the old value from sector tracker could cause it to be reallocated for later entries in a batch write. The idea is to avoid that.

Reply:
valid point. that should be trivial to defer until later.